crates/engine/tree/src/tree/root.rs

```diff
@@ -454,7 +449,10 @@ where
                                 roots_calculated,
                                 "All proofs processed, ending calculation"
                             );
-                            return Ok((current_root, trie_updates));
+                            let mut trie = self.sparse_trie.take().unwrap();
+                            let root = trie.root().unwrap();
```

Comment:
same as above for these unwraps, not sure what the expect message should be here though